osu.Game/Rulesets/Objects/HitObject.cs

```diff
@@ -144,9 +144,10 @@ protected virtual void CreateNestedHitObjects()
 
         /// <summary>
         /// Creates the <see cref="Judgement"/> that represents the scoring information for this <see cref="HitObject"/>.
-        /// May be null.
+        /// Used to decide on drawable object lifetimes.
         /// </summary>
-        public virtual Judgement CreateJudgement() => null;
+        [NotNull]
+        public abstract Judgement CreateJudgement();
```

Comment:
I'd much rather prefer this was `=> new Judgement()` and the `HitWindows` maintained below as in the previous version.